### PR TITLE
Move PhotoCalTask and Colorterm class here from meas_astrom

### PR DESCRIPTION
**Pull request by SimonKrughoff**
In addition to those classes I moved:
- The one photoCal example
- The unit tests

I also copied some test data. It's not a lot of data and I feel much more comfortable
with a local copy that we can rely on, instead of using data from a different package
that might change without warning.

### CODE REVIEW

File: tests/astrometry_net_data/photocal/andConfig.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+root.defaultMagColumn = 'r'
+root.starGalaxyColumn = 'star'
+#root.variableColumn = 'variable'
```
Comment (SimonKrughoff) [inline tests/astrometry_net_data/photocal/andConfig.py]: I'm not exactly sure why we need all these configs.  If we don't need them all, I'd like to remove the unused ones.  I would also like to clean up the commented out code.

**Comment by PaulPrice** [comment]
Do you want to pull the changes to Colorterm from HSC before you move this?

**Comment by SimonKrughoff** [comment]
@PaulPrice I wasn't aware of changes from HSC what are they?

**Comment by PaulPrice** [comment]
Changed Colorterms so it's not a global, and it can now be configured using `Config`.  See https://github.com/HyperSuprime-Cam/meas_astrom/blob/master/python/lsst/meas/photocal/colorterms.py

**Comment by r-owen** [comment]
I would rather make updating Colorterm a separate ticket. It does sound like a useful change.

File: tests/astrometry_net_data/photocal/andConfig.py (hunk 1 above)
Comment (r-owen) [inline tests/astrometry_net_data/photocal/andConfig.py]: I am happy to remove uncommented code, but I don't think it is safe to remove the configs without extensive testing, and the example you give above looks useful.

Comment (r-owen) [inline tests/astrometry_net_data/photocal/andConfig.py]: Never mind. You were right about the andconfig files. Only one of them (the one with no digit in the name) was wanted. I'll remove the others and remove the one commented-out line from the remaining file.

**Comment by PaulPrice** [comment]
Having it another ticket is fine, but you may want to do it before moving packages.

**Comment by jdswinbank** [comment]
This work is long since complete.

**Pull request closed without merging**